### PR TITLE
applications: asset_tracker_v2: send modem_traces on crash

### PR DESCRIPTION
**Pull request by maxd-nordic**
This patch makes use of the modem trace RAM backend to send traces when a crash happened.

### CODE REVIEW

**Comment by maxd-nordic** [comment]
I currently have some trouble getting coredumps to be stored.

File: applications/asset_tracker_v2/overlay-memfault.conf [hunk 1]
```diff
@@ -24,3 +24,7 @@ CONFIG_DEBUG_MODULE=y
 # Increase the event storage size so that all metrics generated by the asset tracker application
 # are reliably sent to the memfault cloud.
 CONFIG_MEMFAULT_EVENT_STORAGE_SIZE=2048
+CONFIG_NRF_MODEM_LIB_TRACE=y
```
Comment (simensrostad) [inline applications/asset_tracker_v2/overlay-memfault.conf:27]: Perhaps we should make a Kconfig option that select these options. `CONFIG_DEBUG_MODULE_MEMFAULT_SEND_MODEM_TRACES_ON_CRASH` (very long, but something similar).

The user can then easily opt out of capturing modem traces by disabling a single option.

File: applications/asset_tracker_v2/src/modules/debug_module.c [hunk 2]
```diff
@@ -33,6 +38,71 @@
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(MODULE, CONFIG_DEBUG_MODULE_LOG_LEVEL);
 
+#if defined(CONFIG_NRF_MODEM_LIB_TRACE_BACKEND_RAM) && defined(CONFIG_MEMFAULT)
```
Comment (simensrostad) [inline applications/asset_tracker_v2/src/modules/debug_module.c:41]: With a new option that I proposed we could replace these options with the new one.

Comment (simensrostad) [inline applications/asset_tracker_v2/src/modules/debug_module.c:41]:
```suggestion
#if defined(CONFIG_DEBUG_MODULE_MEMFAULT_SEND_MODEM_TRACES_ON_CRASH)
```

**Pull request closed without merging**